src/components/album/AlbumChat.js

```diff
@@ -47,19 +74,89 @@ const AlbumChat = ({ apiUrl }) => {
         socket.disconnect();
       }
     };
-  }, [albumId, socket]);
+  }, [albumId, socket, loggedInUserId]);
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
 
   return (
-    <div>
-      <h2>Album Chat</h2>
-      <div>
-        <input
-          type="text"
-          placeholder="Enter message"
-          value={message}
-          onChange={(e) => setMessage(e.target.value)}
-        />
-        <button onClick={handleSendMessage}>Send</button>
+    <div
+      style={{
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+      }}
+    >
+      <Typography variant="h6" component="div" style={{ marginLeft: '20px' }}>
+        Album Chat
+      </Typography>
+      <div
+        style={{
+          flex: 1,
+          padding: '16px',
+          width: '95%',
+          maxHeight: '65vh',
+          overflowY: 'auto',
+        }}
+        ref={messagesContainerRef}
+      >
+        {messages.map((message) => (
+          <div key={uuidv4()}>
```

Comment:
[https://react.dev/learn/rendering-lists](https://react.dev/learn/rendering-lists)

> Similarly, do not generate keys on the fly, e.g. with `key={Math.random()}`. This will cause keys to never match up between renders, leading to all your components and DOM being recreated every time. Not only is this slow, but it will also lose any user input inside the list items. Instead, use a stable ID based on the data.

If you want to generate an id here in the frontend (rather than returning the `ChatMessage` object from the backend which will have an `_id` field), that's totally fine :+1: but we should not generate the ID within the rendered JSX based on the above info from the offical React docs.

Instead, we should generate them in the component outside of the `return (...)` part so that React can keep track of the value between renders. My suggestion would then be to do so when we call `setMessages` on line 60. We can do something like `data.messageKey = uuidv4()`. Or you can use object destructuring to do it on one line if you prefer.